src/Compilers/CSharp/Portable/Parser/Lexer.cs

```diff
@@ -674,7 +674,7 @@ private void ScanSyntaxToken(ref TokenInfo info)
                     break;
 
                 case '<':
-                    if (this.ModeIs(LexerMode.DebuggerSyntax) && TextWindow.PeekChar(1) == '>')
+                    if (this.ModeIs(LexerMode.DebuggerSyntax) && TextWindow.PeekChar(1) == '>' && SyntaxFacts.IsIdentifierStartCharacter(TextWindow.PeekChar(2)))
                     {
                         // For "<>f_AnonymousType", which is an identifier in DebuggerSyntax mode
                         goto case 'a';
```

Comment:
I assumed we did (since the code/comment looked very deliberate), but it looks like removing the special cases for '<' doesn't cause any test failures.

I'm fine with deleting if you can't see any reason for them...